compiler/src/iree/compiler/Codegen/Common/GPU/GPUHeuristics.cpp

```diff
@@ -17,10 +20,88 @@ using llvm::APIntOps::GreatestCommonDivisor;
 
 namespace mlir::iree_compiler {
 
-std::optional<GPUMMASchedule>
+int64_t sharedMemoryUsed(const GPUMMASchedule &schedule, int64_t lhsBitwidth,
+                         int64_t rhsBitwidth) {
+  int64_t tileM = schedule.mSize * schedule.mTileCount * schedule.mWarpCount;
+  int64_t tileN = schedule.nSize * schedule.nTileCount * schedule.nWarpCount;
+  int64_t tileK = schedule.kSize * schedule.kTileCount;
+  return tileM * tileK * lhsBitwidth + tileN * tileK * rhsBitwidth;
+}
+
+bool isValidSchedule(const GPUMatmulShapeType &problem,
+                     const GPUMMASchedule &schedule) {
+  return problem.mSize %
```

Comment:
Can you hoist these 3 into local variables? Will be easier to check under a debugger.